packages/mui-material/test/typescript/hoc-interop.spec.tsx

```diff
@@ -21,22 +21,22 @@ const filledProps = {
 
 // baseline behavior
 <TextField variant="filled" {...filledProps} />;
-// @ts-expect-error
```

Comment:
removed this because didn't understand the reasoning behind expecting typescript error when classes are getting applied.

Reply:
None of the tests in this file using TextField is relevant anymore as TextFieldProps is not a union. I'm not aware of any other component that had such props, so we may remove these tests.

Reply:
I didn't mean to remove the whole file, only the tests using TextField. There are other tests using a Button that are still relevant.

Reply:
ah okay, apologizes for misunderstanding. resolved in this commit (https://github.com/mui/material-ui/pull/36737/commits/954ab7c437f8a3d15e0a98b0ffbc0bbaacd6e7c4) i've removed instances where only `TextFieldProps` is being tested

Reply:
I don't see `TextField` tests removed in `packages/mui-material/test/typescript/hoc-interop.spec.tsx`.

Reply:
Removed textfield tests `[here](https://github.com/mui/material-ui/pull/36737/files#diff-47e3902aef4284face7bfb85ae3fc60fd45cc53eecbe1bff8d5b1e8eb097fb81L23)`, but looks like I have to remove some more. Can you just point me to tests which should be removed, I'll remove those.

Reply:
See the comment on the top of the file. These tests are for High order components (`emotionStyled` and `styled` HoC's) that receive components whose props are union types. Now that `TextFieldProps` is not a union type anymore, we can remove all the tests of the Text Field component in this file.

Reply:
Got it, removed all instances of textfield. Can you verify now

Reply:
We should remove [this](https://github.com/mui/material-ui/pull/36737/files#diff-47e3902aef4284face7bfb85ae3fc60fd45cc53eecbe1bff8d5b1e8eb097fb81L7) comment as well.

Reply:
done